src/main/scala/com/fulcrumgenomics/util/RefFlatSource.scala

```diff
@@ -0,0 +1,210 @@
+                            private[this] val source: Option[{ def close(): Unit }] = None
+                           ) extends Iterable[Gene] with Closeable with LazyLogging {
+  private val genes = {
+
+    // Ensure there is a header for DelimitedDataParser, and if a Picard-style header is used, convert it
+    val _lines: Iterator[String] = {
+      import com.fulcrumgenomics.FgBioDef._
+      val iterator = lines.bufferBetter
+
+      val header = iterator.head
+      if (header.startsWith("geneName")) {
+        iterator
+      }
+      else if (header.startsWith("GENE_NAME")) {
+        // convert the Picard-style header
+        Iterator(header.split('\t').map(RefFlatSource.PicardHeaderMap(_)).mkString("\t")) ++ iterator.drop(1)
+      }
+      else {
+        Iterator(RefFlatSource.Header.mkString("\t")) ++ iterator
+      }
+    }
+
+
+    // Each line is a gene and transcript.  We want to group all transcripts from the same gene.
+    var numDifferentChromsomes = 0
+    var numDifferentStrands    = 0
+    var numTranscripts         = 0
+    val _genes = new DelimitedDataParser(lines=_lines, delimiter='\t').flatMap { row =>
+      val geneName       = row[String]("geneName")
+      val transcriptName = row[String]("name")
+      val contig         = row[String]("chrom")
+      val strand         = row[String]("strand")
+      val exonCount      = row[Int]("exonCount")
+      val exonStarts     = row[String]("exonStarts").split(',').filter(_.nonEmpty).map(_.toInt)
+      val exonEnds       = row[String]("exonEnds").split(',').filter(_.nonEmpty).map(_.toInt)
+
+      if (dict.exists(_.getSequence(contig) == null)) { // skip unrecognized sequences
+        None
+      }
+      else {
+        require(exonStarts.length == exonCount, s"Number of exonStarts does not equal exonCount for $geneName/$transcriptName")
+        require(exonEnds.length == exonCount, s"Number of exonEnds does not equal exonCount for $geneName/$transcriptName")
+
+        // Convert from 0-based half-open to 1-based inclusive intervals.
+        val transcript = Transcript(
+          name     = transcriptName,
+          start    = row[Int]("txStart") + 1,
+          end      = row[Int]("txEnd"),
+          cdsStart = row[Int]("cdsStart") + 1,
+          cdsEnd   = row[Int]("cdsEnd"),
+          exons    = exonStarts.zip(exonEnds).map { case (s, e) => Exon(start=s + 1, end=e) }
+        )
+
+        val gene = Gene(
+          contig         = contig,
+          start          = -1,
+          end            = -1,
+          negativeStrand = strand == "-",
+          name           = geneName,
+          transcripts    = Seq(transcript)
+        )
+
+        Some(gene)
+      }
+    }
+      .toSeq
+      .groupBy(_.name)
+      .map { case (name, _genes) =>
+        val contig         = _genes.head.contig
+        val negativeStrand = _genes.head.negativeStrand
+
+        val transcripts = _genes.flatMap { gene =>
+          require(gene.length() == 1)
```

Comment:
Is this necessary?  It confused the crap out of me, because I initially thought your were asserting that the gene had a single transcript at this point, and that was an odd use of `length()`.  After looking around I see that you're just asserting your two `-1`s above haven't been modified.  Maybe if you want to do that you could assert `start == -1` instead since I think the intent is clearer?